src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+package cromwell.engine.db.slick
+
+import java.util.{Date, UUID}
+
+import com.typesafe.config.Config
+import cromwell.binding._
+import cromwell.binding.types.WdlType
+import cromwell.binding.values.{WdlPrimitive, WdlValue}
+import cromwell.engine._
+import cromwell.engine.backend.Backend
+import cromwell.engine.backend.local.LocalBackend
+import cromwell.engine.db.DataAccess.WorkflowInfo
+import cromwell.engine.db._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.implicitConversions
+
+object SlickDataAccess {
+  implicit class DateToTimestamp(val date: java.util.Date) extends AnyVal {
+    def toTimestamp = new java.sql.Timestamp(date.getTime)
+  }
+}
+
+// TODO: Still needs more refactoring
+
+// Example: Instead of nested .filter(), breakup and compose actions, then use for-comprehension "if" style
+// - Composing: https://github.com/slick/slick/blob/master/slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/TransactionTest.scala
+// - .filter(): http://slick.typesafe.com/doc/3.0.0/queries.html#sorting-and-filtering
+// - for based if: http://slick.typesafe.com/doc/3.0.0/queries.html#monadic-joins
+// While refactoring, ran into an error: "polymorphic expression cannot be instantiated to expected type"
+// Based on google, may be a problem with implicits, requiring explicit specification of some types.
+
+class SlickDataAccess(databaseConfig: Config) extends DataAccess {
+
+  def this() = this(DatabaseConfig.databaseConfig)
+
+  private val IoInput = "INPUT"
+  private val IoOutput = "OUTPUT"
+
+  import SlickDataAccess._
+
+  val dataAccess = new DataAccessComponent(databaseConfig.getString("slick.driver"))
+
+  import dataAccess.driver.api._
+
+  // TODO: Used for flatMap. Should we build another ec instead of using global?
```

Comment:
I might have said this elsewhere (in which case ignore me) but IMO it's fine to use the global context for now. At some point we'll start bogging down the global context and this is the likely candidate for the first place to look at for customization